src/main/java/xbot/common/subsystems/drive/SwerveKinematicsCalculator.java

```diff
@@ -0,0 +1,298 @@
+package xbot.common.subsystems.drive;
+
+import xbot.common.logging.RobotAssertionManager;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class SwerveKinematicsCalculator {
+
+    /*
+    The SwerveKinematicsCalculator is designed to compute the motion of a swerve drive system from a starting position
+     and velocity to a goal position and velocity. It takes into account parameters like maximum acceleration,
+     goal velocity, and maximum velocity to generate a sequence of motion stages, ensuring precise path planning
+     and execution.
+
+    (Breaks an XbotSwervePoint down into smaller nodes containing time, velocity, and acceleration)
+    Note: goal velocity may not always be reached, but in such case will be as close as possible.
+
+    Code involving the quadratic formula IS NOT ROBUST
+
+    And, 99 if not 100% of the math in this calculator is derived from 1D motion physics formulas! (Need Algebra)
+
+    TODO: Naming may have inconsistencies... maybe.
+    */
+
+    final double startPosition;
```

Comment:
⭐ ⭐   what are the units of these? Could a Units class be used to be more specific or could the unit be put in the variable name otherwise? Is it meters? is it a ratio 0-1?